src/vs/code/electron-main/app.ts

```diff
@@ -420,6 +422,91 @@ export class CodeApplication extends Disposable {
 		}
 	}
 
+	/**
+	 * Add filtering of web requests.
+	 */
+	private registerWebRequestFilter() {
+		const { defaultSession } = session;
+
+		// Block all SVG requests from unsupported origins
+		const supportedSvgSchemes = new Set([Schemas.file, Schemas.vscodeFileResource, Schemas.vscodeRemoteResource, 'devtools']);
+
+		// But allow them if the are made from inside an webview
+		const isSafeFrame = (requestFrame: WebFrameMain | undefined): boolean => {
+			for (let frame: WebFrameMain | null | undefined = requestFrame; frame; frame = frame.parent) {
+				if (frame.url.startsWith(`${Schemas.vscodeWebview}://`)) {
+					return true;
+				}
+			}
+			return false;
+		};
+
+		const isSvgRequestFromSafeContext = (details: Electron.OnBeforeRequestListenerDetails | Electron.OnHeadersReceivedListenerDetails): boolean => {
+			return details.resourceType === 'xhr' || isSafeFrame(details.frame);
+		};
+
+		const isAllowedVsCodeFileRequest = (details: Electron.OnBeforeRequestListenerDetails) => {
+			if (!details.frame || !this.windowsMainService) {
+				return false;
+			}
+
+			for (const window of this.windowsMainService.getWindows()) {
+				if (window.win && details.frame.routingId === window.win.webContents.mainFrame.routingId) {
```

Comment:
`routingId` is only unique within a process, this handler will be triggered for requests from multiple processes. So you would also need to compare `processId` to make sure the frame matches.

Also, web worker extension host uses an iframe to host the contents and I think it uses `vscode-file` scheme for the top-level frame request. It will blocked by this handler since it is not the main frame in the window. We might either allow request to that web worker extension host initiator files or we can switch it to use a separate scheme (this will make it a cross-origin worker and will require https://github.com/microsoft/vscode/issues/87282#issuecomment-917539104 workaround)

Reply:
In my testing, the web worker iframe has the same processId as the main window. I believe this is because they are from the same origin?

Do you think this is a safe assumption? If not, I could also allow requests from iframes that are themselves served from `vscode-file` urls